main/src/library/Int64.flix

```diff
@@ -266,6 +266,23 @@ namespace Int64 {
     @Time(1) @Space(1)
     pub def flipBit(x: Int64, position: Int32): Int64 = x ^^^ (1i64 <<< position)
 
+    ///
+    /// Returns the integer binary logarithm of `x`.
+    ///
+    @Time(1) @Space(1)
+    pub def log2(x: Int64): Int32 = highestOneBitPosition(x)
+    ///
+    /// Returns the factorial of `x`.
+    ///
+    @Time(factorial(x)) @Space(1)
+    pub def factorial(x: Int64): Int64 = factorialHelper(x, 1i64)
+
+    ///
+    /// Helper function for `factorial`.
+    ///
+    def factorialHelper(x: Int64, acc: Int64): Int64 =
```

Comment:
One quick comment is that we now prefer to use local defs. See e.g. how it is done in `List`.

Reply:
Thanks for your feedback, I've applied a fix as 756966e01.